vault/resource_identity_entity.go

```diff
@@ -242,28 +247,59 @@ func identityEntityIDPath(id string) string {
 }
 
 func readIdentityEntityPolicies(client *api.Client, entityID string) ([]interface{}, error) {
-	resp, err := readIdentityEntity(client, entityID)
+	resp, err := readIdentityEntity(client, entityID, false)
 	if err != nil {
 		return nil, err
 	}
-	if resp == nil {
-		return nil, fmt.Errorf("error IdentityEntity %s does not exist", entityID)
-	}
 
 	if v, ok := resp.Data["policies"]; ok && v != nil {
 		return v.([]interface{}), nil
 	}
 	return make([]interface{}, 0), nil
 }
 
-// May return nil if entity does not exist
-func readIdentityEntity(client *api.Client, entityID string) (*api.Secret, error) {
+func readIdentityEntity(client *api.Client, entityID string, retry bool) (*api.Secret, error) {
 	path := identityEntityIDPath(entityID)
-	log.Printf("[DEBUG] Reading Entity %s from %q", entityID, path)
+	log.Printf("[DEBUG] Reading Entity %q from %q", entityID, path)
+
+	return readEntity(client, path, retry)
+}
+
+func readEntity(client *api.Client, path string, retry bool) (*api.Secret, error) {
+	log.Printf("[DEBUG] Reading Entity from %q", path)
+
+	var err error
+	if retry {
+		client, err = client.Clone()
+		if err != nil {
+			return nil, err
+		}
+
+		maxRetries := client.MaxRetries()
+		if maxRetries == 0 {
```

Comment:
Is this just to be safe? Since we're setting `clientConfig.MaxRetries = DefaultMaxHTTPRetries` in the provider configuration, would this value ever be 0? Is this something users can pass through a TF config and set to be something else?

Reply:
Unless the value doesn't get copied during cloning

Reply:
Yes, this is just to ensure that we always retry, since [max_retries](https://registry.terraform.io/providers/hashicorp/vault/latest/docs#max_retries) is a provider configurable

Reply:
Thanks for clarifying